src/Umbraco.Web.BackOffice/Controllers/ContentController.cs

```diff
@@ -2277,139 +2275,135 @@ public ActionResult<DomainSave> PostSaveLanguageAndDomains(DomainSave model)
             }
         }
 
+        // Validate node
         IContent? node = _contentService.GetById(model.NodeId);
-
         if (node == null)
         {
             HttpContext.SetReasonPhrase("Node Not Found.");
             return NotFound("There is no content node with id {model.NodeId}.");
         }
 
-        EntityPermission? permission =
-            _userService.GetPermissions(_backofficeSecurityAccessor.BackOfficeSecurity?.CurrentUser, node.Path);
-
-
+        // Validate permissions on node
+        EntityPermission? permission = _userService.GetPermissions(_backofficeSecurityAccessor.BackOfficeSecurity?.CurrentUser, node.Path);
         if (permission?.AssignedPermissions.Contains(ActionAssignDomain.ActionLetter.ToString(), StringComparer.Ordinal) == false)
         {
             HttpContext.SetReasonPhrase("Permission Denied.");
             return BadRequest("You do not have permission to assign domains on that node.");
         }
 
         model.Valid = true;
-        IDomain[]? domains = _domainService.GetAssignedDomains(model.NodeId, true)?.ToArray();
+
+        IDomain[] assignedDomains = _domainService.GetAssignedDomains(model.NodeId, true).ToArray();
         ILanguage[] languages = _localizationService.GetAllLanguages().ToArray();
-        ILanguage? language = model.Language > 0 ? languages.FirstOrDefault(l => l.Id == model.Language) : null;
 
-        // process wildcard
-        if (language != null)
+        // Process language
+        var language = model.Language > 0 ? languages.FirstOrDefault(l => l.Id == model.Language) : null;
```

Comment:
We should use explicit type here 😁 
```suggestion
        ILanguage? language = model.Language > 0 ? languages.FirstOrDefault(l => l.Id == model.Language) : null;
```